src/libs/Navigation/AppNavigator/usePreloadFullScreenNavigators.ts

```diff
@@ -0,0 +1,142 @@
+
+    navigation.preload(NAVIGATORS.WORKSPACE_SPLIT_NAVIGATOR, {
+        screen: focusedWorkspaceRoute.name,
+        params: focusedWorkspaceRoute.params,
+    });
+}
+
+function preloadReportsTab(navigation: PlatformStackNavigationProp<AuthScreensParamList>) {
+    const lastSearchNavigator = navigation.getState().routes.findLast((route) => route.name === NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR);
+    const lastSearchNavigatorState = lastSearchNavigator && lastSearchNavigator.key ? getPreservedNavigatorState(lastSearchNavigator?.key) : undefined;
+    const lastSearchRoute = lastSearchNavigatorState?.routes.findLast((route) => route.name === SCREENS.SEARCH.ROOT);
+
+    if (lastSearchRoute) {
+        const {q, ...rest} = lastSearchRoute.params as SearchFullscreenNavigatorParamList[typeof SCREENS.SEARCH.ROOT];
+        const queryJSON = buildSearchQueryJSON(q);
+        if (queryJSON) {
+            const query = buildSearchQueryString(queryJSON);
+            navigation.preload(NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR, {screen: SCREENS.SEARCH.ROOT, params: {q: query, ...rest}});
+        }
+    }
+
+    navigation.preload(NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR, {screen: SCREENS.SEARCH.ROOT, params: {q: buildCannedSearchQuery()}});
+}
+
+function preloadAccountTab(navigation: PlatformStackNavigationProp<AuthScreensParamList>, subscriptionPlan: ValueOf<typeof CONST.POLICY.TYPE> | null) {
+    if (!getIsNarrowLayout()) {
+        const settingsTabState = getSettingsTabStateFromSessionStorage();
+
+        if (!settingsTabState) {
+            navigation.preload(NAVIGATORS.SETTINGS_SPLIT_NAVIGATOR, {screen: SCREENS.SETTINGS.PROFILE.ROOT, params: {}});
+            return;
+        }
+
+        const screenName = findFocusedRoute(settingsTabState)?.name as keyof SettingsSplitNavigatorParamList | undefined;
+
+        if ((!subscriptionPlan && screenName === SCREENS.SETTINGS.SUBSCRIPTION.ROOT) || !screenName) {
+            navigation.preload(NAVIGATORS.SETTINGS_SPLIT_NAVIGATOR, {screen: SCREENS.SETTINGS.PROFILE.ROOT, params: {}});
+            return;
+        }
+
+        navigation.preload(NAVIGATORS.SETTINGS_SPLIT_NAVIGATOR, {
+            screen: screenName,
+        });
+        return;
+    }
+    navigation.preload(NAVIGATORS.SETTINGS_SPLIT_NAVIGATOR, {screen: SCREENS.SETTINGS.ROOT});
+}
+
+function preloadInboxTab(navigation: PlatformStackNavigationProp<AuthScreensParamList>) {
+    const payload = getIsNarrowLayout() ? {screen: SCREENS.HOME} : {screen: SCREENS.REPORT};
+    navigation.preload(NAVIGATORS.REPORTS_SPLIT_NAVIGATOR, payload);
+}
+
+/**
+ * Hook that preloads all fullscreen navigators except the current one.
+ * This helps improve performance by preloading navigators that might be needed soon.
+ */
+function usePreloadFullScreenNavigators(navigation: PlatformStackNavigationProp<AuthScreensParamList>, fullscreenTabName: keyof typeof NAVIGATION_TABS) {
+    const state = navigation.getState();
+    const {preloadedRoutes} = state;
```

Comment:
I am not sure how I feel about such code in the body of component. Maybe some memo, effect or smth?